src/planscape/impacts/services.py

```diff
@@ -0,0 +1,30 @@
+from typing import Type
+from django.db import transaction
+from impacts.models import TreatmentPlan, TreatmentPlanStatus
+from planning.models import Scenario
+from django.contrib.auth.models import AbstractUser
+from actstream import action
+
+TreatmentPlanType = Type[TreatmentPlan]
+ScenarioType = Type[Scenario]
+UserType = Type[AbstractUser]
+
+
+@transaction.atomic()
+def create_treatment_plan(
+    scenario: ScenarioType,
+    name: str,
+    created_by: UserType,
+) -> TreatmentPlanType:
+    # question: should we add a constraint on
+    # treament plan to prevent users from creating
+    # treamentplans with for the same scenario with the
+    # same name?
```

Comment:
I think this is probably a good idea, although it adds a little bit of frontend work, I think. But better to implement more constraints now than trying to do it later.